dev/devicelab/lib/framework/framework.dart

```diff
@@ -179,16 +180,14 @@ class _TaskRunner {
             if (result is TaskResultCheckProcesses) {
               result = TaskResult.failure('This test leaked dart processes');
             }
-            final bool killed = await killProcess(info.pid);
+            final bool killed = Process.killPid(info.pid, ProcessSignal.sigkill);
```

Comment:
This doesn't work on Windows.

Reply:
(And Windows historically has been our biggest culprit for this kind of problem, especially since leaked processes can lock files something else wants/needs to delete).

Last time I brought this up with anyone from the VM team, it seems like a fundamental limitation on the API offered by windows. Hence the delegation to `taskkill.exe` which generally does a better job (but is not really suitable for use by the Dart VM as API surface IMO).

Reply:
roger. Is there an issue tracking this?

Reply:
https://github.com/dart-lang/sdk/issues/40759

And related https://github.com/dart-lang/sdk/issues/47675

Reply:
neither of those seem that problematic here?

Reply:
I added those in a comment (and continued to use `taskkill.exe`), but they don't seem like reasons we fundamentally can't use this API on Windows.

Reply:
The API always returns false, and will not forcefully terminate the process. What we want here is forceful termination.

Reply:
> The API always returns false, and will not forcefully terminate the process. What we want here is forceful termination.

Is there a bug that tracks this issue?

Reply:
@aam see the linked bugs in https://github.com/flutter/flutter/pull/96320#discussion_r781453699

Reply:
@dnfield I looked at those - they seem to be about wrong return value or lack of waiting, they don't mention that killPid doesn't actually terminate given process.

Reply:
Ahhh that part is a bit harder - but my recollection is that on Windows I would sometimes see a process that would only terminate if the `/f` flag was passed to `taskkill.exe`, and Dart had no equivalent way of doing the force part.

Reply:
Dart also has no way of waiting for the process to terminate, so we don't know if we're continuing execution after the process has finished or not.

Reply:
Yeah, we could make `killPid` to wait for the process to exit. Some processes might take longer to exit, which will block this call too. Alternatively if caller insists on waiting it supposedly could call `killPid` until it returns `false`.

Would be nice if there was a repro for when `killPid` didn't actually work and you had to resort to shelling `taskkill.exe`